// react-js-loading-shimmer/dist/index.js
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var PropTypes = _interopDefault(require('prop-types'));

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css = ".loading-shimmer{background-color:#eee;background-image:-webkit-gradient(linear,left top,right top,from(#eee),color-stop(#f5f5f5),to(#eee));background-image:linear-gradient(90deg,#eee,#f5f5f5,#eee);background-repeat:no-repeat;background-size:100% 100%;border-radius:4px;display:inline-block;line-height:1;width:100%}.loading-shimmer-animation{-webkit-animation:progress 1.2s ease-in-out infinite;animation:progress 1.2s ease-in-out infinite}@-webkit-keyframes progress{0%{background-position:-200px 0}to{background-position:calc(200px + 100%) 0}}@keyframes progress{0%{background-position:-200px 0}to{background-position:calc(200px + 100%) 0}}";
styleInject(css);

var classCallCheck = function (instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
};

var createClass = function () {
  function defineProperties(target, props) {
    for (var i = 0; i < props.length; i++) {
      var descriptor = props[i];
      descriptor.enumerable = descriptor.enumerable || false;
      descriptor.configurable = true;
      if ("value" in descriptor) descriptor.writable = true;
      Object.defineProperty(target, descriptor.key, descriptor);
    }
  }

  return function (Constructor, protoProps, staticProps) {
    if (protoProps) defineProperties(Constructor.prototype, protoProps);
    if (staticProps) defineProperties(Constructor, staticProps);
    return Constructor;
  };
}();









var inherits = function (subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
};











var possibleConstructorReturn = function (self, call) {
  if (!self) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return call && (typeof call === "object" || typeof call === "function") ? call : self;
};

/**
 * @class Shimmer
 */

var Shimmer = function (_Component) {
  inherits(Shimmer, _Component);

  function Shimmer() {
    classCallCheck(this, Shimmer);
    return possibleConstructorReturn(this, (Shimmer.__proto__ || Object.getPrototypeOf(Shimmer)).apply(this, arguments));
  }

  createClass(Shimmer, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          className = _props.className,
          Sheight = _props.Sheight;

      var class_name = "loading-shimmer loading-shimmer-animation " + (className ? className : "");
      return React__default.createElement('span', { className: class_name,
        style: { height: Sheight ? Sheight : "16px" } });
    }
  }]);
  return Shimmer;
}(React.Component);

Shimmer.propTypes = {
  className: PropTypes.string,
  Sheight: PropTypes.string
};

module.exports = Shimmer;
